File: content-solutions/hash/subarray-sum.js
function solution(nums, k) {
  const seen = new Map([[0, 1]]);
  let total = 0;
  let answer = 0;
  for (const x of nums) {
    total += x;
    answer += seen.get(total - k) ?? 0;
    seen.set(total, (seen.get(total) ?? 0) + 1);
  }
  return answer;
}

File: content-solutions/hash/subarray-sum.py
def solution(nums, k):
    seen = {0: 1}
    total = 0
    answer = 0
    for x in nums:
        total += x
        answer += seen.get(total - k, 0)
        seen[total] = seen.get(total, 0) + 1
    return answer
